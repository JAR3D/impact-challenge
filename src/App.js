import { useEffect, useState } from "react";

import { fetchBeers } from "./api";

import BeerList from "./components/BeerList/BeerList";
import BeerDetails from "./components/BeerDetails/BeerDetails";
import AddBeerForm from "./components/AddBeerForm/AddBeerForm";

import classes from "./App.module.css";

function App() {
  const [beerList, setBeerList] = useState([]);
  const [selected, setSelected] = useState(null);

  const getBeers = async () => {
    const { data } = await fetchBeers();
    setBeerList(data);
  };

  const addBeer = (beer) => {
    setBeerList([...beerList, beer]);
  };

  useEffect(() => {
    getBeers();
  }, []);

  return (
    <div className={classes.wrapper}>
      <h1>My Beer Cellar</h1>
      <AddBeerForm addBeer={addBeer} />
      <hr />
      <div className={classes.action}>
        <BeerList list={beerList} onClick={setSelected} />
        {selected !== null ? (
          <BeerDetails
            name={beerList[selected].name}
            description={beerList[selected].description}
            labels={beerList[selected].labels}
          />
        ) : (
          <p className={classes["none-selected"]}>Please select a beer</p>
        )}
      </div>
    </div>
  );
}

export default App;
