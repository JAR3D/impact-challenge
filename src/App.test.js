import { render, screen } from "@testing-library/react";
import App from "./App";

describe("App", () => {
  it("should render loading", () => {
    const { baseElement } = render(<App />);
    expect(baseElement).toMatchSnapshot();
    expect(screen.getByText(/loading/i)).toBeInTheDocument();
  });
});
