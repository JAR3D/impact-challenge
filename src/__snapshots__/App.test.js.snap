// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render loading 1`] = `
<body>
  <div>
    <div
      class="wrapper"
    >
      <h1>
        My Beer Cellar
      </h1>
      <form
        class="form"
      >
        <div>
          <label
            for="name"
          >
            Name:
          </label>
          <input
            id="name"
            name="name"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="description"
          >
            Description:
          </label>
          <input
            id="description"
            name="description"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="label-url"
          >
            Label url:
          </label>
          <input
            id="label-url"
            name="label-url"
            type="text"
            value=""
          />
        </div>
        <button
          type="submit"
        >
          Add Beer
        </button>
      </form>
      <hr />
      <div
        class="action"
      >
        <p>
          Loading...
        </p>
        <p
          class="none-selected"
        >
          Please select a beer
        </p>
      </div>
    </div>
  </div>
</body>
`;
